src/ui/server/controllers/insurance/policy-and-export/single-contract-policy/validation/rules/contract-completion-date.test.ts

```diff
@@ -0,0 +1,196 @@
+import { add, getDate, getMonth, getYear, sub } from 'date-fns';
+import contractCompletionDateRules from './contract-completion-date';
+import { FIELD_IDS, PRODUCT } from '../../../../../../constants';
+import { ERROR_MESSAGES } from '../../../../../../content-strings';
+import generateValidationErrors from '../../../../../../helpers/validation';
+
+const {
+  INSURANCE: {
+    POLICY_AND_EXPORTS: {
+      CONTRACT_POLICY: {
+        REQUESTED_START_DATE,
+        SINGLE: { CONTRACT_COMPLETION_DATE },
+      },
+    },
+  },
+} = FIELD_IDS;
+
+const {
+  INSURANCE: {
+    POLICY_AND_EXPORTS: {
+      CONTRACT_POLICY: {
+        SINGLE: { [CONTRACT_COMPLETION_DATE]: ERROR_MESSAGE },
+      },
+    },
+  },
+} = ERROR_MESSAGES;
+
+describe('controllers/insurance/policy-and-export/single-contract-policy/validation/rules/contract-completion-date', () => {
+  const mockErrors = {
+    summary: [],
+    errorList: {},
+  };
+
+  describe('when day field is not provided', () => {
+    it('should return validation error', () => {
+      const mockSubmittedData = {
+        [`${CONTRACT_COMPLETION_DATE}-month`]: '1',
+        [`${CONTRACT_COMPLETION_DATE}-year`]: '2022',
+      };
+
+      const result = contractCompletionDateRules(mockSubmittedData, mockErrors);
+
+      const expected = generateValidationErrors(CONTRACT_COMPLETION_DATE, ERROR_MESSAGE.IS_EMPTY, mockErrors);
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when month field is not provided', () => {
+    it('should return validation error', () => {
+      const mockSubmittedData = {
+        [`${CONTRACT_COMPLETION_DATE}-day`]: '10',
+        [`${CONTRACT_COMPLETION_DATE}-year`]: '2022',
+      };
+
+      const result = contractCompletionDateRules(mockSubmittedData, mockErrors);
+
+      const expected = generateValidationErrors(CONTRACT_COMPLETION_DATE, ERROR_MESSAGE.IS_EMPTY, mockErrors);
+
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('when year field is not provided', () => {
+    it('should return validation error', () => {
+      const mockSubmittedData = {
+        [`${CONTRACT_COMPLETION_DATE}-day`]: '10',
+        [`${CONTRACT_COMPLETION_DATE}-month`]: '1',
+      };
+
+      const result = contractCompletionDateRules(mockSubmittedData, mockErrors);
+
+      const expected = generateValidationErrors(CONTRACT_COMPLETION_DATE, ERROR_MESSAGE.IS_EMPTY, mockErrors);
+
+      expect(result).toEqual(expected);
+    });
+  });
+
```

Comment:
Can we please add a test case when neither of all three (day, month and year) are provided.

Reply:
We don't need to 😄 We have logic to check if all 3 fields are provided and have tests to make sure that if any single one of these fields is missing, an error is returned. 100% test coverage